@xen-orchestra/rest-api/src/rest-api/rest-api.mts

```diff
@@ -2,10 +2,14 @@ import type { XoApp } from './rest-api.type.mjs'
 
 export class RestApi {
   authenticateUser: XoApp['authenticateUser']
+  getObject: XoApp['getObject']
+  getObjectsByType: XoApp['getObjectsByType']
   runWithApiContext: XoApp['runWithApiContext']
 
   constructor(xoApp: XoApp) {
     this.authenticateUser = (credentials, userData, opts) => xoApp.authenticateUser(credentials, userData, opts)
+    this.getObject = (id, type) => xoApp.getObject(id, type)
+    this.getObjectsByType = (type, opts) => xoApp.getObjectsByType(type, opts)
```

Comment:
would it work with real methods defined on the class and setting xoa as a private member of Rest Api ?